base/array.jl

```diff
@@ -71,8 +71,36 @@ Union type of [`Vector{T}`](@ref) and [`Matrix{T}`](@ref).
 """
 const VecOrMat{T} = Union{Vector{T}, Matrix{T}}
 
+"""
+    DenseArray{T, N} <: AbstractArray{T,N}
+
+`N`-dimensional dense array with elements of type `T`, often used to represent
+a mathematical vector, matrix, or higher dimensional tensor.
```

Comment:
I don't think we need the
> often used to represent a mathematical vector, matrix, or higher dimensional tensor.

part, these are used just as often for other things, no?

Reply:
I think it's useful to have those terms in the documentation to help with searching cross-referencing.

Reply:
Sure, but there's nothing special about `DenseArray` that makes it particularly better at being a mathematical vector/matrix/tensor.